cypress/support/commands.js

```diff
@@ -0,0 +1,42 @@
+import { TOTP } from 'totp-generator'
+
+Cypress.Commands.add('login', () => {
+  cy.visit('/')
+
+  cy.get('nav.navbar').then((navbar) => {
+    if (navbar.find('a:contains("Log in with GitHub")').length > 0) {
+      cy.get('a:contains("Log in with GitHub")').click()
+    }
+  })
+
+  cy.origin('github.com', { args: { otpContainer: TOTP.generate(Cypress.env('OAUTH_OTP_SECRET')) } }, ({ otpContainer }) => {
```

Comment:
Nice work getting this all working, I would have given up well before this and just attempted to run the cypress tests against a local version with the authentication flow mocked/switched off, but it's definitely more valuable having a real login flow as part of it

That said, particularly given we're relying on GitHub as a 3rd party, be aware that for a real app we might admit there's a risk that GitHub starts blocking us here (e.g. by using Captcha)